src/Pos/Wallet/Web/Mode.hs

```diff
@@ -0,0 +1,185 @@
+import           Pos.Util.JsonLog              (jsonLogReal)
+import           Pos.Util.TimeWarp             (CanJsonLog (..))
+import           Pos.Wallet.Redirect           (MonadBlockchainInfo (..),
+                                                MonadUpdates (..),
+                                                applyLastUpdateWebWallet,
+                                                blockchainSlotDurationWebWallet,
+                                                connectedPeersWebWallet,
+                                                localChainDifficultyWebWallet,
+                                                networkChainDifficultyWebWallet,
+                                                waitForUpdateWebWallet)
+import           Pos.Wallet.SscType            (WalletSscType)
+import           Pos.Wallet.Web.Server.Sockets (ConnectionsVar)
+import           Pos.Wallet.Web.State.State    (WalletState)
+import           Pos.Wallet.Web.Tracking       (MonadWalletTracking (..),
+                                                syncOnImportWebWallet,
+                                                syncWSetsAtStartWebWallet,
+                                                txMempoolToModifierWebWallet)
+import           Pos.WorkMode                  (RealModeContext)
+
+data WalletWebModeContext = WalletWebModeContext
+    { wmcWalletState     :: !WalletState
+    , wmcWebSockets      :: !ConnectionsVar
+    , wmcRealModeContext :: !(RealModeContext WalletSscType)
+    }
+
+makeLensesFor
+    [ ("wmcRealModeContext", "wmcRealModeContextL")
+    , ("wmcWalletState",     "wmcWalletStateL")
+    , ("wmcWebSockets",      "wmcWebSocketsL") ]
+    ''WalletWebModeContext
+
+instance {-# OVERLAPPABLE #-} HasLens tag (RealModeContext WalletSscType) r => HasLens tag WalletWebModeContext r where
+    lensOf = wmcRealModeContextL . lensOf @tag
+
+instance HasLens WalletState WalletWebModeContext WalletState where
+    lensOf = wmcWalletStateL
+
+instance HasLens ConnectionsVar WalletWebModeContext ConnectionsVar where
+    lensOf = wmcWebSocketsL
+
+data WWEB
+
+type WalletWebMode = ExecMode WWEB
+
+type instance ExecModeM WWEB =
+    Mtl.ReaderT WalletWebModeContext Production
+
+unWalletWebMode :: ExecMode WWEB a -> ExecModeM WWEB a
+unWalletWebMode = unExecMode
+
+instance WithPeerState WalletWebMode where
+    getPeerState = getPeerStateReal
+    clearPeerState = clearPeerStateReal
+    getAllStates = getAllStatesReal
+
+instance MonadSlotsData WalletWebMode where
+    getSystemStart = getSystemStartReal
+    getSlottingData = getSlottingDataReal
+    waitPenultEpochEquals = waitPenultEpochEqualsReal
+    putSlottingData = putSlottingDataReal
+
+instance MonadSlots WalletWebMode where
+    getCurrentSlot = getCurrentSlotReal
+    getCurrentSlotBlocking = getCurrentSlotBlockingReal
+    getCurrentSlotInaccurate = getCurrentSlotInaccurateReal
+    currentTimeSlotting = currentTimeSlottingReal
```

Comment:
Do you have ideas how to avoid writing it, because it's already defined for `RealMode` and it's basically a super mode of `RealMode`?